heron/schedulers/src/java/com/twitter/heron/scheduler/local/LocalScheduler.java

```diff
@@ -261,12 +262,15 @@ public boolean onUpdate(Scheduler.UpdateTopologyRequest request) {
   @Override
   public void addContainers(Set<PackingPlan.ContainerPlan> containers) {
     synchronized (processToContainer) {
+      Set<Integer> containerIds = new HashSet<>();
+      containerIds.addAll(processToContainer.values());
       for (PackingPlan.ContainerPlan container : containers) {
-        if (processToContainer.containsKey(container.getId())) {
+        if (containerIds.contains(container.getId())) {
```

Comment:
That's a great observation @windie. If we were running production topologies with this class that added many containers I'd agree that this is a concern. In this case though, this is a local scheduler used for local development which we wouldn't expect to need to add more than max 2-3 containers to ever. For that reason I think think we should favor simplicity and code readability over the performance aspect.